sound/soc/sof/intel/hda-dai.c

```diff
@@ -309,6 +324,19 @@ static int __maybe_unused hda_dai_trigger(struct snd_pcm_substream *substream, i
 			dev_err(sdev->dev, "%s: failed to clean up link DMA\n", __func__);
 			return ret;
 		}
+
+		/*
+		 * With IPC4 the pipelines (including the dai widget)) are not
+		 * freed up on stop/suspend.
+		 * If the stream is re-started after stop instead of closing it
+		 * then we need to keep the link DMA reserved.
+		 */
```

Comment:
It would be worth commenting the link DMA flow in a bit more detail since this is complex (if not already detailed in code).